truffle/src/main/java/org/jruby/truffle/core/string/StringNodes.java

```diff
@@ -1449,6 +1459,321 @@ private int prevCharHead(Encoding enc, byte[]bytes, int p, int s, int end) {
         }
     }
 
+    @CoreMethod(names = "scrub_internal", optional = 1, needsBlock = true)
```

Comment:
You can pass it as a normal argument, like `prim(arg1, block)` and in the surrounding method capture it with `&block`.